### DISCUSSION TITLE
Unable to resolve module @react-native-firebase/auth

### DISCUSSION THREAD

**Discussion by sea-sol** [invertase/react-native-firebase#5699] [Q&A] [answered]
when I try to run sample code from common document site.[https://rnfirebase.io/auth/usage#listening-to-authentication-state](url)
I am getting this issue

error: Error: Unable to resolve module @react-native-firebase/auth from prodject path\sendotp.js: @react-native-firebase/auth could not be found within the project or in these directories:
  node_modules

If you are sure the module exists, try these steps:
 1. Clear watchman watches: watchman watch-del-all
 2. Delete node_modules and run yarn install
 3. Reset Metro's cache: yarn start --reset-cache
 4. Remove the cache: rm -rf /tmp/metro-*

I tried all this, but getting same issue 

this is my package.json 

"dependencies": {
    "@pspatel/react-native-app-intro": "^1.1.0",
    "@react-native-community/checkbox": "^0.5.8",
    "@react-native-community/masked-view": "^0.1.11",
    "@react-native-firebase/app": "^12.7.5",
    "axios": "^0.21.1",
    "form-data": "^4.0.0",
    "moment": "^2.29.1",
    "radio-buttons-react-native": "^1.0.4",
    "react": "17.0.2",
    "react-hooks-global-state": "^1.0.2",
    "react-native": "0.65.1",
    "react-native-autofocus": "0.0.7",
    "react-native-calendars": "^1.1266.0",
    "react-native-collapsible": "^1.6.0",
    "react-native-elements": "^3.4.2",
    "react-native-gesture-handler": "^1.10.3",
    "react-native-image-crop-picker": "^0.36.2",
    "react-native-image-picker": "^4.0.6",
    "react-native-loading-spinner-overlay": "^2.0.0",
    "react-native-maps": "^0.29.3",
    "react-native-modal": "^13.0.0",
    "react-native-paper": "^4.9.2",
    "react-native-reanimated": "2.3.0-alpha.2",
    "react-native-safe-area-context": "^3.3.0",
    "react-native-screens": "^3.6.0",
    "react-native-select-dropdown": "^1.1.0",
    "react-native-snackbar": "^2.4.0",
    "react-native-splash-screen": "^3.2.0",
    "react-native-unique-id": "^2.0.0",
    "react-native-vector-icons": "^8.1.0",
    "react-navigation": "^4.4.4",
    "react-navigation-material-bottom-tabs": "^2.3.4",
    "react-navigation-stack": "^2.10.4",
    "react-navigation-tabs": "^2.11.1",
    "react-redux": "^7.2.4",
    "redux": "^4.1.1"
  },

why sample code makes issue for me?
please help me

**Comment by mikehardy** [comment, marked as answer]
You must install a module prior to using it. You have not installed the auth module, just as the error says.

**Reply by sea-sol** [reply]
Oh my hero. why did I forgot just install auth module? it's really simple reason. It seems I thought auth module is in app module. anyway, thank you so much. and sorry for late reply.
